### PR TITLE
824 empty image

### PR DESCRIPTION
**Pull request by BernardHan**
This PR added the empty image into all "events showing" pages, also added loading logic in those pages so that they show loading spinner instead of empty image when loading.

Loading component is also refined in term of style and style variable passing.

#824

### CODE REVIEW

File: src/scenes/Activities/ResultSetting/store.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { queryAllOracles } from '../../../network/graphql/queries';
 const INIT_VALUES = {
   loaded: false, // loading state?
   loadingMore: false, // for laoding icon?
+  loadingFirst: false,
```
Comment (dwalintukan) [inline src/scenes/Activities/ResultSetting/store.js]: remove

File: src/scenes/Activities/Withdraw/index.js [hunk 2]
```diff
@@ -15,8 +15,11 @@ export default class Withdraw extends Component {
   }
 
   render() {
-    const { list, loadMore, loadingMore, loading } = this.props.store.activities.withdraw;
-    if (loading) return <Loading />;
+    const { list, loadMore, loadingMore, loadingFirst } = this.props.store.activities.withdraw;
+    const loadProps = {
```
Comment (dwalintukan) [inline src/scenes/Activities/Withdraw/index.js]: JSS

File: src/scenes/Activities/Finalize/index.js [hunk 3]
```diff
@@ -15,8 +15,11 @@ export default class Finalize extends Component {
   }
 
   render() {
-    const { list, loadMore, loadingMore, loading } = this.props.store.activities.finalize;
-    if (loading) return <Loading />;
+    const { list, loadMore, loadingMore, loadingFirst } = this.props.store.activities.finalize;
+    const loadProps = {
+      marginTop: '10rem',
+    };
+    if (loadingFirst) return <Loading {...loadProps} />;
```
Comment (dwalintukan) [inline src/scenes/Activities/Finalize/index.js]: `if (!loaded) ...`

File: src/scenes/Activities/Finalize/store.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ import { queryAllOracles } from '../../../network/graphql/queries';
 const INIT_VALUES = {
   loaded: false, // loading state?
   loadingMore: false, // for laoding icon?
+  loadingFirst: false,
```
Comment (dwalintukan) [inline src/scenes/Activities/Finalize/store.js]: remove, use `loaded`. `loadingFirst` is the opposite of `loaded`

File: src/scenes/Activities/ResultSetting/index.js [hunk 5]
```diff
@@ -15,8 +15,11 @@ export default class ResultSetting extends Component {
   }
 
   render() {
-    const { list, loadMore, loadingMore, loading } = this.props.store.activities.resultSetting;
-    if (loading) return <Loading />;
+    const { list, loadMore, loadingMore, loadingFirst } = this.props.store.activities.resultSetting;
+    const loadProps = {
+      marginTop: '10rem',
+    };
+    if (loadingFirst) return <Loading {...loadProps} />;
```
Comment (dwalintukan) [inline src/scenes/Activities/ResultSetting/index.js]: `if (!loaded) ...`

File: src/scenes/Activities/Withdraw/index.js [hunk 6]
```diff
@@ -15,8 +15,11 @@ export default class Withdraw extends Component {
...
+      marginTop: '10rem',
+    };
+    if (loadingFirst) return <Loading {...loadProps} />;
```
Comment (dwalintukan) [inline src/scenes/Activities/Withdraw/index.js]: `if (!loaded) ...`

File: src/scenes/Activities/Withdraw/store.js [hunk 7]
```diff
@@ -8,6 +8,7 @@ import { queryAllVotes, queryAllTopics } from '../../../network/graphql/queries'
 const INIT_VALUES = {
   loaded: false, // loading state?
   loadingMore: false, // for laoding icon?
+  loadingFirst: false,
```
Comment (dwalintukan) [inline src/scenes/Activities/Withdraw/store.js]: remove

File: src/scenes/AllEvents/index.js [hunk 8]
```diff
@@ -36,8 +24,11 @@ export default class AllEvents extends Component {
   }
 }
 
-const Events = observer(({ allEvents: { list, loadMoreEvents, loading, loadingMore } }) => {
-  if (loading) return <Loading />;
+const Events = observer(({ allEvents: { list, loadMoreEvents, loadingFirst, loadingMore } }) => {
+  const loadProps = {
```
Comment (dwalintukan) [inline src/scenes/AllEvents/index.js]: JSS

File: src/scenes/AllEvents/index.js [hunk 9]
```diff
@@ -36,8 +24,11 @@ export default class AllEvents extends Component {
...
+    marginTop: '10rem',
+  };
+  if (loadingFirst) return <Loading {...loadProps} />;
```
Comment (dwalintukan) [inline src/scenes/AllEvents/index.js]: `if (!loaded) ...`

File: src/scenes/BotCourt/index.js [hunk 10]
```diff
@@ -15,8 +15,11 @@ export default class BotCourt extends Component {
   }
 
   render() {
-    const { list, loadMore, loadingMore, loading } = this.props.store.botCourt;
-    if (loading) return <Loading />;
+    const { list, loadMore, loadingMore, loadingFirst } = this.props.store.botCourt;
+    const loadProps = {
```
Comment (dwalintukan) [inline src/scenes/BotCourt/index.js]: JSS

File: src/scenes/BotCourt/index.js [hunk 11]
```diff
@@ -15,8 +15,11 @@ export default class BotCourt extends Component {
...
+      marginTop: '10rem',
+    };
+    if (loadingFirst) return <Loading {...loadProps} />;
```
Comment (dwalintukan) [inline src/scenes/BotCourt/index.js]: `if (!loaded) ...`

File: src/scenes/QtumPrediction/index.js [hunk 12]
```diff
@@ -36,8 +36,11 @@ export default class QtumPrediction extends Component {
   }
 
   render() {
-    const { list, loadMore, loadingMore, loading } = this.props.store.qtumPrediction;
-    if (loading) return <Loading />;
+    const { list, loadMore, loadingMore, loadingFirst } = this.props.store.qtumPrediction;
+    const loadProps = {
+      marginTop: '10rem',
+    };
+    if (loadingFirst) return <Loading {...loadProps} />;
```
Comment (dwalintukan) [inline src/scenes/QtumPrediction/index.js]: `if (!loaded) ...`

File: src/scenes/QtumPrediction/store.js [hunk 13]
```diff
@@ -8,6 +8,7 @@ import Oracle from '../../stores/models/Oracle';
 const INIT_VALUES = {
   loaded: false, // loading state?
   loadingMore: false, // for laoding icon?
+  loadingFirst: false,
```
Comment (dwalintukan) [inline src/scenes/QtumPrediction/store.js]: remove

File: src/scenes/QtumPrediction/store.js [hunk 14]
```diff
@@ -51,21 +53,26 @@ export default class QtumPredictionStore {
 
   @action
   loadFirst = async () => {
+    this.loadingFirst = true;
```
Comment (dwalintukan) [inline src/scenes/QtumPrediction/store.js]: `Object.assign(this, INIT_VALUES)` to reset everything

File: src/scenes/Search/index.js [hunk 15]
```diff
@@ -146,9 +150,12 @@ export default class Search extends Component {
 }
 
 const NoResult = () => (
-  <Typography variant="body1">
-    <FormattedMessage id="search.emptySearchResult" defaultMessage="Oops, your search has no results." />
-  </Typography>
+  <Row>
+    <Row><img src="/images/empty.svg" alt="empty placeholder" /></Row>
+    <Row><Typography variant="title">
```
Comment (dwalintukan) [inline src/scenes/Search/index.js]: nit: separate components on diff lines

File: src/scenes/Search/styles.js [hunk 16]
```diff
@@ -20,15 +20,11 @@ const styles = (theme) => ({
...
+    display: 'grid',
     justifyContent: 'center',
     alignItems: 'center',
     width: '100%',
-    margin: theme.paddingSm,
-  },
-  searching: {
-    width: '5rem',
-    height: '10rem',
+    margin: '20px',
```
Comment (dwalintukan) [inline src/scenes/Search/styles.js]: use closest theme.padding.

File: src/stores/AllEventsStore.js [hunk 17]
```diff
@@ -21,6 +22,7 @@ export default class {
   @observable list = INIT_VALUES.list
   @observable hasMoreTopics = INIT_VALUES.hasMoreTopics
   @observable hasMoreOracles = INIT_VALUES.hasMoreOracles
+  @observable loadingFirst = INIT_VALUES.loadingFirst
```
Comment (dwalintukan) [inline src/stores/AllEventsStore.js]: remove

File: src/stores/BotCourtStore.js [hunk 18]
```diff
@@ -19,6 +20,7 @@ export default class BotCourtStore {
   @observable list = INIT_VALUES.list
   @observable hasMore = INIT_VALUES.hasMore
   @observable skip = INIT_VALUES.skip
+  @observable loadingFirst = INIT_VALUES.loadingFirst
```
Comment (dwalintukan) [inline src/stores/BotCourtStore.js]: remove

File: src/components/EventListLoading/styles.js [hunk 19]
```diff
@@ -4,12 +4,7 @@ const styles = () => ({
     justifyContent: 'center',
     alignItems: 'center',
     width: '100%',
-    margin: 20,
-  },
-  loading: {
-    width: '5rem',
-    height: '5rem',
-    marginTop: 360,
+    margin: '20px',
```
Comment (dwalintukan) [inline src/components/EventListLoading/styles.js]: use 20 (no 'px' unless needed), its preferred in JSS

File: src/components/InfiniteScroll/index.js [hunk 20]
```diff
@@ -1,17 +1,21 @@
...
+import styles from './styles';
 
 const messages = defineMessages({
   loadMoreMsg: {
     id: 'load.more',
     defaultMessage: 'loading more',
   },
 });
-
+@withWidth()
```
Comment (dwalintukan) [inline src/components/InfiniteScroll/index.js]: nit: add line above

File: src/components/InfiniteScroll/index.js [hunk 21]
```diff
@@ -49,26 +53,39 @@ export default class InfiniteScroll extends Component {
...
+          </Typography>
+        </Row>
+        {ui.location === Routes.QTUM_PREDICTION &&
+        <Row>
+          <Button
+            size={width === 'xs' ? 'small' : 'medium'}
+            color="primary"
+            onClick={createEvent.open}
+          >
+            <FormattedMessage id="create.dialogTitle" defaultMessage="CREATE AN EVENT" />
```
Comment (dwalintukan) [inline src/components/InfiniteScroll/index.js]: I changed the text for dialogTitle to `"Create Event"`

File: src/components/InfiniteScroll/styles.js [hunk 22]
```diff
@@ -0,0 +1,11 @@
+const styles = () => ({
+  row: {
+    display: 'grid',
+    justifyContent: 'center',
+    alignItems: 'center',
+    width: '100%',
+    marginTop: '20px',
```
Comment (dwalintukan) [inline src/components/InfiniteScroll/styles.js]: 20

File: src/components/InfiniteScroll/index.js [hunk 23]
```diff
@@ -49,26 +53,39 @@ export default class InfiniteScroll extends Component {
...
+    const { data, spacing, loadingMore, store, width } = this.props;
+    const { createEvent, ui } = store;
+    return data.length > 0 ? (
+      <Grid container spacing={spacing}>
+        {data}
+        {loadingMore && <LoadingMore />}
       </Grid>
+    ) : (
+      <Row>
+        <Row><img src="/images/empty.svg" alt="empty placeholder" /></Row>
```
Comment (dwalintukan) [inline src/components/InfiniteScroll/index.js]: nit: your choice, but seeing as the duplicate code in multiple places its prob best to make its own component:

**NoItemsPlaceholder**
```
const messages = defineMessages({
  // messages of the different empty placeholder messages
});

const NoItemsPlaceholder = ({ intl, myMessageId }) => (
  <div>
    <Row><img src="/images/empty.svg" alt="empty placeholder" /></Row>
    <Row>
       <Typography variant="title">
         {intl.formatMessage(message.myMessageId)}
       </Typography>
    </Row>
  </div>
);

export default (NoItemsPlaceholder);
```

File: src/scenes/Activities/ActivityHistory/index.js [hunk 24]
```diff
@@ -104,9 +107,14 @@ export default class ActivityHistory extends Component {
                 />
               </Table>
             ) : (
-              <Typography variant="body1">
-                <FormattedMessage id="str.emptyTxHistory" defaultMessage="You do not have any transactions right now." />
-              </Typography>
+              <Row>
+                <Row><img src="/images/empty.svg" alt="empty placeholder" /></Row>
```
Comment (dwalintukan) [inline src/scenes/Activities/ActivityHistory/index.js]: good place for reusable component

File: src/scenes/Search/index.js [hunk 25]
```diff
@@ -146,9 +146,12 @@ export default class Search extends Component {
 }
 
 const NoResult = () => (
-  <Typography variant="body1">
-    <FormattedMessage id="search.emptySearchResult" defaultMessage="Oops, your search has no results." />
-  </Typography>
+  <Row>
+    <Row><img src="/images/empty.svg" alt="empty placeholder" /></Row>
```
Comment (dwalintukan) [inline src/scenes/Search/index.js]: good place for reuseable component

File: src/components/EventListLoading/styles.js (hunk 19 above)
Comment (BernardHan) [reply src/components/EventListLoading/styles.js]: I tried without px, but it won't work as I expected

File: src/components/InfiniteScroll/index.js (hunk 23 above)
Comment (BernardHan) [reply src/components/InfiniteScroll/index.js]: I will component it

File: src/components/InfiniteScroll/index.js [hunk 26]
```diff
@@ -49,26 +53,39 @@ export default class InfiniteScroll extends Component {
...
-        {this.props.loadingMore && <LoadingMore />}
+    const { data, spacing, loadingMore, store, width } = this.props;
+    const { createEvent, ui } = store;
+    return data.length > 0 ? (
+      <Grid container spacing={spacing}>
+        {data}
+        {loadingMore && <LoadingMore />}
       </Grid>
+    ) : (
+      <Row>
```
Comment (ippudo) [inline src/components/InfiniteScroll/index.js]: Why use Row? Using Grid please

File: src/scenes/Activities/ActivityHistory/index.js [hunk 27]
```diff
@@ -104,9 +107,14 @@ export default class ActivityHistory extends Component {
                 />
               </Table>
             ) : (
-              <Typography variant="body1">
-                <FormattedMessage id="str.emptyTxHistory" defaultMessage="You do not have any transactions right now." />
-              </Typography>
+              <Row>
```
Comment (ippudo) [inline src/scenes/Activities/ActivityHistory/index.js]: Move the whole placeholder out of Grid. It will leads inconsistent of position with other pages.

**Pull request merged**